marconi/app/Main.hs

```diff
@@ -133,16 +149,25 @@ getDatums (BlockInMode (Block (BlockHeader slotNo _ _) txs) _) = concatMap extra
       let hashes = assocs . fst $ scriptDataFromCardanoTxBody txBody
        in map (slotNo,) hashes
 
+isMambaTxOut :: MambaAddresses -> C.TxOut C.CtxTx era -> Bool
+isMambaTxOut mambaAddresses (C.TxOut addr _ _) =
+    toShelleyAddr addr `elem` fmap toShelleyAddr  mambaAddresses
```

Comment:
@koslambrou applying `Cardano.Api.toShelleyAddr` here for every hit is wasteful. 
The datatype, `Shelley.Addr StandardCrypto` is not exported so I couldn't do the conversion prior to storing it in a list. I am sure there is a way, I just couldn't find it :(
Is there a better way of doing this?

Reply:
Yeah... do you actually need to do `toShelleyAddr`. Why not `addr `elem` mambaAddresses?

Reply:
it seems that I do as I get build error:
`    • Couldn't match type ‘NonEmpty.NonEmpty
                             (C.AddressInEra C.ShelleyEra)’
                     with ‘a0 -> era’
      Expected type: a0 -> era
        Actual type: MambaAddresses
    • In the first argument of ‘fmap’, namely ‘mambaAddresses’
      In the second argument of ‘elem’, namely ‘fmap mambaAddresses’
      In the expression: addr `elem` fmap mambaAddresses
    • Relevant bindings include
        addr :: C.AddressInEra era (bound at app/Main.hs:153:38)
        isMambaTxOut :: MambaAddresses -> C.TxOut C.CtxTx era -> Bool
          (bound at app/Main.hs:153:1)
    |
154 |     addr `elem` fmap mambaAddresses
    |                      ^^^^^^^^^^^^^`